Lib/encodings/rot_13.py

```diff
@@ -8,21 +8,30 @@
 
 import codecs
 
+def _check_str(input):
+    if not isinstance(input, str):
+        raise TypeError(
+            "a str object is required, not {}".format(type(input).__name__))
+
 ### Codec APIs
 
 class Codec(codecs.Codec):
     def encode(self, input, errors='strict'):
+        _check_str(input)
         return (input.translate(rot13_map), len(input))
```

Comment:
What if just use `str.translate(input, rot13_map)`?

Reply:
I love this approach. It avoids making the code ugly.